### PR TITLE
[SPARK-36335] Add local-cluster docs to developer-tools.md

### PR DESCRIPTION
**Pull request by yutoacts**
Document "local-cluster" mode on developer-tools.md.

Related links:
https://github.com/apache/spark/pull/33568
https://github.com/apache/spark/pull/33537
https://issues.apache.org/jira/browse/SPARK-36335
https://issues.apache.org/jira/browse/SPARK-595

### CODE REVIEW

**Comment by srowen** [comment]
This doesn't quite feel like the right place to document this. How about docs/spark-standalone.md in the main Spark project docs?

**Comment by HyukjinKwon** [comment]
I actually suggested to avoid documenting it in the main docs because local-cluster is test-only mode. But I am fine with doing it in `docs/spark-standalone.md` too. Looks like Tom wants to have it in main docs too.

**Comment by srowen** [comment]
I see, if this is really intended as a developer tool, this would be the right place. The very old SPARK-595 thread suggests it isn't totally for testing.

File: site/sitemap.xml [hunk 1]
```diff
@@ -907,10 +907,6 @@
   <loc>https://spark.apache.org/mllib/</loc>
   <changefreq>weekly</changefreq>
 </url>
-<url>
```
Comment (srowen) [inline site/sitemap.xml]: (Revert this change)

**Comment by tgravescs** [comment]
so I guess it isn't that big of deal because we don't do document it now and doesn't seem to have been a big problem, but I assume this issue was filed for a reason and my concern is that people know about and use local-cluster mode so why not just clarify what its for.  there are 2 things, one is obscure it by not documenting it which works for some people, but it doesn't work for others that know about it or find it but don't know its for unit testing only.  If we document all the run modes in a common places seems like it would be easier to find for users.  
While it does run like standalone mode it doesn't make sense to me to put under standalone mode docs, at least I wouldn't think to go there to look for it.

**Comment by tgravescs** [comment]
Note if others disagree, I'm fine with leaving here in developer docs. I would rather see it go in common docs where we describe all run modes or in developer docs to obscure from users.

**Comment by HyukjinKwon** [comment]
My only concern about documenting in the main docs is that it happens to force us to investigate/document it together whenever a cluster related feature (like archive, resource profile, etc.), and then it gives the dev some more overhead of investigation, for example, the one instance in the main document (https://spark.apache.org/docs/latest/configuration.html#custom-resource-scheduling-and-configuration-overview).

If we'll document once, and explicitly say there's no gurnatee on such features since that's a test only mode, I'm fine with doing it in the main docs too.

File: site/sitemap.xml (hunk 1 above)
Comment (yutoacts) [reply site/sitemap.xml]: Reverted. Thank you.

**Comment by HyukjinKwon** [comment]
I am fine either way, no big deal. I will defer to @srowen and @tgravescs.

**Comment by yutoacts** [comment]
Thanks for the suggestions. If it goes in main docs, should it be documented in `docs/spark-standalone.md` or `docs/submitting-applications.md` (as my initial PR: https://github.com/apache/spark/pull/33537)? IMHO documenting it in `docs/spark-standalone.md` might confusing people with standalone mode.

File: site/developer-tools.html [hunk 2]
```diff
@@ -793,6 +793,18 @@ <h4>In Spark unit tests</h4>
 The platform-specific paths to the profiler agents are listed in the 
 <a href="http://www.yourkit.com/docs/80/help/agent.jsp">YourKit documentation</a>.</p>
 
+<p><a name="local-cluster"></a></p>
+<h3>Local-cluster mode</h3>
+
+<p>When launching applications with spark-submit, besides options in 
+<a href="https://spark.apache.org/docs/latest/submitting-applications.html#master-urls">Master URLs</a>
+, set local-cluster option to emulate a distributed cluster in a single JVM.</p>
```
Comment (tgravescs) [inline site/developer-tools.html:801]: shouldn't we explicitly say this is for unit testing only?

Comment (yutoacts) [reply site/developer-tools.html:801]: I thought it is not necessary as it's on a developer-tools page but I'm totally fine with explicitly saying that here.
BTW if local-cluster mode ends up written in main documents (https://github.com/apache/spark/pull/33537), should it still be documented here?

**Comment by yutoacts** [comment]
It ended up as https://github.com/apache/spark/pull/33537.
